### PR TITLE
Cleanup Ole32 enums

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Cleanup Ole32 enums (extracted from #1961)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2127)

### CODE REVIEW

File: src/Common/src/Interop/Ole32/Interop.CALLCONV.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+internal static partial class Interop
+{
+    internal static partial class Ole32
+    {
+        public enum CALLCONV : int
+        {
+            FASTCALL = 0,
```
Comment (zsd4yr) [inline src/Common/src/Interop/Ole32/Interop.CALLCONV.cs]: please define explicitly

File: src/Common/src/Interop/Ole32/Interop.FUNCFLAGS.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+using System;
+
+internal static partial class Interop
+{
+    internal static partial class Ole32
+    {
+        [Flags]
+        public enum FUNCFLAGS : ushort
+        {
+            FRESTRICTED = 0x0001,
```
Comment (zsd4yr) [inline src/Common/src/Interop/Ole32/Interop.FUNCFLAGS.cs]: I know we want to match these but it is a shame for readability that all of these needlessly start with F

File: src/Common/src/Interop/Ole32/Interop.FUNCKIND.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+internal static partial class Interop
+{
+    internal static partial class Ole32
+    {
+        public enum FUNCKIND : int
+        {
+            VIRTUAL,
```
Comment (zsd4yr) [inline src/Common/src/Interop/Ole32/Interop.FUNCKIND.cs]: please explicitly define

File: src/Common/src/Interop/Ole32/Interop.INVOKEKIND.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+using System;
+
+internal static partial class Interop
+{
+    internal static partial class Ole32
+    {
+        [Flags]
+        public enum INVOKEKIND : int
+        {
+            FUNC = 1,
```
Comment (zsd4yr) [inline src/Common/src/Interop/Ole32/Interop.INVOKEKIND.cs]: nit: I think these are supposed to be hex. I realize the values are identical

File: src/Common/src/Interop/Ole32/Interop.TYPEKIND.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+internal static partial class Interop
+{
+    internal static partial class Ole32
+    {
+        public enum TYPEKIND : int
```
Comment (zsd4yr) [inline src/Common/src/Interop/Ole32/Interop.TYPEKIND.cs]: Should we refer to these by TYPEKIND or TKIND?

File: src/Common/src/Interop/Ole32/Interop.FUNCFLAGS.cs (hunk 2 above)
Comment (hughbe) [reply src/Common/src/Interop/Ole32/Interop.FUNCFLAGS.cs]: I can remove if you think?

File: src/Common/src/Interop/Ole32/Interop.TYPEKIND.cs (hunk 5 above)
Comment (hughbe) [reply src/Common/src/Interop/Ole32/Interop.TYPEKIND.cs]: https://github.com/tpn/winsdk-10/blob/9b69fd26ac0c7d0b83d378dba01080e93349c2ed/Include/10.0.16299.0/um/OAIdl.Idl#L458-L468

I'm following the naming from the header:
```
typedef [v1_enum] enum tagTYPEKIND {
    TKIND_ENUM = 0,
    TKIND_RECORD,
    TKIND_MODULE,
    TKIND_INTERFACE,
    TKIND_DISPATCH,
    TKIND_COCLASS,
    TKIND_ALIAS,
    TKIND_UNION,
    TKIND_MAX                   /* end of enum marker */
} TYPEKIND;

```

File: src/Common/src/Interop/Ole32/Interop.FUNCFLAGS.cs (hunk 2 above)
Comment (zsd4yr) [reply src/Common/src/Interop/Ole32/Interop.FUNCFLAGS.cs]: I am sort of torn; I think it is better to match the headers unfortunately...

image

File: src/Common/src/Interop/Ole32/Interop.TYPEKIND.cs (hunk 5 above)
Comment (zsd4yr) [reply src/Common/src/Interop/Ole32/Interop.TYPEKIND.cs]: OH I saw a duplicate in the src

**Comment by zsd4yr** [comment]
@msftbot please merge if @RussKie approves

**Pull request merged**